### ISSUE TITLE
dynamic variables do not support interactive commands

### ISSUE THREAD

**Issue by marco-m** [go-task/task#1087]
Hello!

Consider the following Taskfile:
```
version: '3'

tasks:
  env:
    cmds:
      - echo "{{.GREETING}}"
    env:
      GREETING:
        sh: passwd
  vars:
    cmds:
      - echo "{{.GREETING}}"
    vars:
      GREETING:
        sh: passwd
```

Where we use an interactive command to set the value of the dynamic variables, both in the case of `vars:` and in the case of `env:`.

I am using `passwd` as example of interactive command because it is the first one I can think of that is present on any installation and that allows me to reproduce the problem. In real-world, one would be using a more reasonable interactive command.

If I run task, I get the following (slightly different but, at the core, same problem). Task terminates immediately, without waiting for the user to send input to the command.

```
$ task vars
passwd: conversation failure
task: [vars] echo "Changing password for marco."
Changing password for marco.
```

and

```
$ task env
passwd: conversation failure
task: [env] echo ""
```

thanks!

- Task version: 3.22
- Operating System: macOS, Linux

**Comment by andreynering** [comment]
Hi @marco-m,

The truth is that this is by design. Allowing that could over complicate things. Nowadays we cache variables, and there are open questions on how these variables would deal with that behavior. Should we run that only once, or every time that variable is invoked?

We'd also have to deal with concurrency, for example. Running that while other tasks are also printing stuff to STDOUT would be messy. Also, what to do when a TTY is not available (CI)?

In short: this needs more thought. Maybe a whole new mechanism needs to be thought to cover this use case instead of allowing that on a dynamic var.

If allowing as you requested, it could be a better idea to wait for the next major version that should do other changes on how variable works, including on how we handle caching.

**Comment by marco-m** [comment]
Hello @andreynering,

Thanks for the explanation. I understand it is not obvious what to do and how.

Before opening, I was browsing the other issues and saw the ones about the caching. My 2 cents is that caching should be removed completely, because as a user I do not expect it and, if I want it, I think it can be obtained by setting the variable as a global, instead than in a given task.

Regarding concurrency, yeah, I see the problem. As a data point, consider that it happens already today, because you could have an interactive command invoked in a `cmd`:
```yaml
tasks:
  foo:
    cmds:
      - passwd
```

thanks again!

**Comment by trulede** [comment]
Closing stale issue. There are other issues tracking interactive usage (i.e. bubble-tea).
